dds/DCPS/TypeObject.cpp

```diff
@@ -0,0 +1,557 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DCPS/DdsDcps_pch.h" //Only the _pch include should start with DCPS/
+
+#include "Hash.h"
+#include "TypeObject.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+MinimalMemberDetail::MinimalMemberDetail(const std::string& name)
+{
+  unsigned char result[16];
+  DCPS::MD5Hash(result, name.c_str(), name.size());
+
+  std::memcpy(name_hash, result, sizeof name_hash);
+}
+
+
+TypeIdentifierPtr makeTypeIdentifier(const TypeObject& type_object)
+{
+  ACE_Message_Block buff(find_size(type_object));
+  DCPS::Serializer ser(&buff, DCPS::Encoding::KIND_XCDR2_PLAIN, DCPS::ENDIAN_LITTLE);
```

Comment:
So I'm at the point where I have to start making changes to TypeObject.cpp to bring it inline with my changes to the serialization interface. In particular I'm going to have to pull these args are out into a `Encoding` object to share with the size function. Out of curiosity, is `ENDIAN_LITTLE` a placeholder or is this one of the places where byte order is fixed?

Reply:
It is defined as little-endian in the spec.